lib/editor/tunnel.js

```diff
@@ -57,9 +58,24 @@ class EditorTunnel {
         info(`Connecting editor tunnel to ${forgeWSEndpoint}`)
 
         // * Enable Device Editor (Step 8) - (device->forge:WS) Initiate WS connection (with token)
+        const headers = {
+            'x-access-token': this.options.token
+        }
+        if (this.affinity) {
+            headers.cookie = `FFSESSION=${this.affinity}`
+        }
         const socket = newWsConnection(forgeWSEndpoint, {
-            headers: {
-                'x-access-token': this.options.token
+            headers
+        })
+        socket.on('upgrade', (evt) => {
+            if (evt.headers && evt.headers['set-cookie']) {
+                const cookies = evt.headers['set-cookie']
+                cookies.forEach(cookie => {
```

Comment:
@hardillb Are we certain that `evt.headers['set-cookie']` is always going to be an array? I don't readily have the means to verify this locally.

Reply:
I'll double check the WS code, but if I can't find a definitive answer I'll add some defensive code